### PR TITLE
Improve completion label details display

### PR DESCRIPTION
**Pull request by ahlinc**
This PR improves completion label details display by separating trait and alias info from the `label` LSP field to an additional existing `label_detail` field. Changes look like the following:

### Before

Screenshot from 2023-11-24 12-03-08

### After

Screenshot from 2023-11-24 12-21-57

_All existing tests are passed without any changes in test themselves logic._

### CODE REVIEW

**Review by lnicola — APPROVED** [review]

File: crates/ide-completion/src/tests.rs [hunk 1]
```diff
@@ -150,16 +150,29 @@ fn render_completion_list(completions: Vec<CompletionItem>) -> String {
     fn monospace_width(s: &str) -> usize {
         s.chars().count()
     }
-    let label_width =
-        completions.iter().map(|it| monospace_width(&it.label)).max().unwrap_or_default().min(22);
+    let label_width = completions
+        .iter()
+        .map(|it| {
+            monospace_width(&it.label)
+                + monospace_width(it.label_detail.as_deref().unwrap_or_default())
```
Comment (lnicola) [inline crates/ide-completion/src/tests.rs:157]: Should we add 1 here for the space?

File: crates/ide-completion/src/item.rs [hunk 2]
```diff
@@ -454,10 +456,17 @@ impl Builder {
         if let [import_edit] = &*self.imports_to_add {
             // snippets can have multiple imports, but normal completions only have up to one
             if let Some(original_path) = import_edit.original_path.as_ref() {
-                label = SmolStr::from(format!("{label} (use {})", original_path.display(db)));
+                label_detail = Some(SmolStr::from(format!(
```
Comment (lnicola) [inline crates/ide-completion/src/item.rs:459]: This is a bit hard to follow, but I don't have any suggestions :shrug:.

Comment (ahlinc) [reply crates/ide-completion/src/item.rs]: @lnicola I rewrote logic to avoid reassignments and force pushed it the PR branch.

Comment (ahlinc) [reply crates/ide-completion/src/item.rs]: But I don't know would @bors see my latest changes and take them into account.

Comment (ahlinc) [reply crates/ide-completion/src/item.rs]: @lnicola As I understand @bors will complete this PR as it saw it on the https://github.com/rust-lang/rust-analyzer/commit/f8bf1dc15d770642b252ed6e35bcfb84ef074fda commit.
Can I open an another PR with a small fix that force pushed here?

**Comment by lnicola** [comment]
I think it's using the new version now: https://github.com/rust-lang/rust-analyzer/commit/87337283235bb4056fb8f8e06127d7e44e58322c.

**Comment by lnicola** [comment]
image

**Comment by hecatia-elegua** [comment]
~I think this might have broken #14775.~
~For structs it still seems to work though.~

edit: okay no, that feature only works if the alias starts with an uppercase letter for some reason.

**Comment by lnicola** [comment]
Consider filing a new issue.
